File: PI_PERSO/WEB_SYMFONY/Reclamations/MobilePush/integration --finale--/WebTemplate/src/Entity/Produit.php
<?php

namespace App\Entity;
use Symfony\Component\HttpFoundation\File\File;
use Vich\UploaderBundle\Mapping\Annotation as Vich;

use Doctrine\ORM\Mapping as ORM;

/**
 * Produit
 *
 * @ORM\Table(name="produit", indexes={@ORM\Index(name="test", columns={"idcat"})})
 * @ORM\Entity
 */
class Produit
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     *
     * @ORM\Column(name="CodeProduit", type="string", length=255, nullable=false)
     */
    private $codeproduit;

    /**
     * @var string
     *
     * @ORM\Column(name="Des", type="string", length=255, nullable=false)
     */
    private $des;

    /**
     * @var int
     *
     * @ORM\Column(name="QteStock", type="integer", nullable=false)
     */
    private $qtestock;

    /**
     * @var int
     *
     * @ORM\Column(name="QteMin", type="integer", nullable=false)
     */
    private $qtemin;

    /**
     * @var int
     *
     * @ORM\Column(name="PrixUnitaire", type="integer", nullable=false)
     */
    private $prixunitaire;

    /**
     * @var string
     *
     * @ORM\Column(name="idUnite", type="string", length=255, nullable=false)
     */
    private $idunite;

    /**
     * @var string
     *
     * @ORM\Column(name="image", type="string", length=255, nullable=false)
     */
    private $image;

    /**
     * @var string
     *
     * @ORM\Column(name="idcat", type="string", length=255, nullable=false)
     */
    private $idcat;

    /**
     * @Vich\UploadableField(mapping="image", fileNameProperty="imageName")
     * 
     * @var File
     */
    private $imageFile;  
    


    public function setImageFile(?File $imageFile = null): void
    {
        $this->imageFile = $imageFile;
    }
    public function getImageFile(): ?File
    {
        return $this->imageFile;
    }
    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCodeproduit(): ?string
    {
        return $this->codeproduit;
    }

    public function setCodeproduit(string $codeproduit): self
    {
        $this->codeproduit = $codeproduit;

        return $this;
    }

    public function getDes(): ?string
    {
        return $this->des;
    }

    public function setDes(string $des): self
    {
        $this->des = $des;

        return $this;
    }

    public function getQtestock(): ?int
    {
        return $this->qtestock;
    }

    public function setQtestock(int $qtestock): self
    {
        $this->qtestock = $qtestock;

        return $this;
    }

    public function getQtemin(): ?int
    {
        return $this->qtemin;
    }

    public function setQtemin(int $qtemin): self
    {
        $this->qtemin = $qtemin;

        return $this;
    }

    public function getPrixunitaire(): ?int
    {
        return $this->prixunitaire;
    }

    public function setPrixunitaire(int $prixunitaire): self
    {
        $this->prixunitaire = $prixunitaire;

        return $this;
    }

    public function getIdunite(): ?string
    {
        return $this->idunite;
    }

    public function setIdunite(string $idunite): self
    {
        $this->idunite = $idunite;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getIdcat(): ?string
    {
        return $this->idcat;
    }

    public function setIdcat(string $idcat): self
    {
        $this->idcat = $idcat;

        return $this;
    }



}

File: PI_Perso_Reclamations/MOBILE_CodenameOne_Reclamations/src/com/mycompany/myapp/entities/Produit.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author ashre
 */
public class Produit {
    private int id; 
    private String Codeproduit;
    private String Des;
    private int QteStock;
    private int QteMin;
    private int PrixUnitaire;
    private String idUnite;
    private String image;
    private String idcat;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getCodeproduit() {
        return Codeproduit;
    }

    public void setCodeproduit(String Codeproduit) {
        this.Codeproduit = Codeproduit;
    }

    public String getDes() {
        return Des;
    }

    public void setDes(String Des) {
        this.Des = Des;
    }

    public int getQteStock() {
        return QteStock;
    }

    public void setQteStock(int QteStock) {
        this.QteStock = QteStock;
    }

    public int getQteMin() {
        return QteMin;
    }

    public void setQteMin(int QteMin) {
        this.QteMin = QteMin;
    }

    public int getPrixUnitaire() {
        return PrixUnitaire;
    }

    public void setPrixUnitaire(int PrixUnitaire) {
        this.PrixUnitaire = PrixUnitaire;
    }

    public String getIdUnite() {
        return idUnite;
    }

    public void setIdUnite(String idUnite) {
        this.idUnite = idUnite;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getIdcat() {
        return idcat;
    }

    public void setIdcat(String idcat) {
        this.idcat = idcat;
    }
    
}
